x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/FieldPermissions.java

```diff
@@ -72,7 +71,7 @@ public final class FieldPermissions implements Accountable, CacheKey {
     private final long ramBytesUsed;
 
     /** Constructor that does not enable field-level security: all fields are accepted. */
-    public FieldPermissions() {
+    private FieldPermissions() {
         this(new FieldPermissionsDefinition(null, null), Automatons.MATCH_ALL);
     }
```

Comment:
Make this constructor private since it is only really useful for building the static `FieldPermissions.DEFAULT` variable. Its usages have been replaced by directly using the variable. This change touched a few files.